### PR TITLE
Change absolute path returned by getPathToModule() to relative path

### PR DESCRIPTION
**Pull request by zcfan**
**Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Not sure how to proper test&document this change, help me improvement please if they are necessary for this PR.

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
#2215

**What is the new behavior?**
Generated require calls is relative paths now.

**Does this PR introduce a breaking change?**
- [ ] Yes
- [x] No

**Other information**:

### CODE REVIEW

**Comment by Kovensky** [comment]
Looks like this broke module resolution 😕

If this code works, the only tests that should break are places that check output paths.

**Comment by zcfan** [comment]
Thanks for reviewing, I'll checkout again closely.

File: lib/node/NodeSourcePlugin.js [hunk 1]
```diff
@@ -37,33 +38,39 @@ NodeSourcePlugin.prototype.apply = function(compiler) {
 			if(localOptions.process) {
 				var processType = localOptions.process;
 				parser.plugin("expression process", function() {
-					return ModuleParserHelpers.addParsedVariable(this, "process", "require(" + JSON.stringify(getPathToModule("process", processType)) + ")");
+					return ModuleParserHelpers.addParsedVariable(this, "process", "require(" +
+					    JSON.stringify(path.relative(this.state.module.context, getPathToModule("process", processType))) + ")");
```
Comment (sokra) [inline lib/node/NodeSourcePlugin.js]: Could you extract your change into a function? And in this function change `\` to `/` for windows.

Comment (zcfan) [inline lib/node/NodeSourcePlugin.js]: Didn't noticed that, thank you for point it out. I'll work on it tonight.
BTW, should I keep other changes that cause test errors?
I can eliminate few of them by modifying dependent packages, but it seems that changing webpack codebase to fit these change is beyond my current ability. T_T

Comment (sokra) [inline lib/node/NodeSourcePlugin.js]: You need to prefix `./` to the relative path to fix the test errors.

**Comment by sokra** [comment]
Could apply the same change to the following too?

* `expression Buffer`
* `expression setImmediate`
* `expression clearImmediate`

File: lib/node/NodeSourcePlugin.js (hunk 1 above)
Comment (zcfan) [inline lib/node/NodeSourcePlugin.js]: Ok, I'm sure I can eliminate every test error by write the declaration of `Buffer`, `setImmediate`, `clearImmediate` before their usage in `webpack/node_modules/buffer/index.js` and `webpack/node_modules/timers-browserify/main.js`

**Comment by sokra** [comment]
And it would be great to reduce the Code Duplication a bit...

File: lib/node/NodeSourcePlugin.js [hunk 2]
```diff
@@ -24,6 +25,11 @@ NodeSourcePlugin.prototype.apply = function(compiler) {
 			return require.resolve("node-libs-browser/mock/empty");
 		} else return module;
 	}
+
+	function getStringOfRelativePath(context, pathToModule) {
+		return JSON.stringify(path.relative(context, pathToModule).replace(/\\/g, '/'))
```
Comment (sokra) [inline lib/node/NodeSourcePlugin.js]: prefix with `./` if:

* it's not already prefixed with `../`
* it's not an absolute path i. e. returned by `path.relative("C:\\a", "D:\\b")`

Here it's already done this way: https://github.com/webpack/webpack/blob/9c5c6df64f7c53f014be058fa18afde03fe6b77f/lib/NodeStuffPlugin.js#L114-L117

Maybe we could merge these code fragments into a common module.

**Comment by zcfan** [comment]
@sokra Yes, but it could cause test errors I didn't find a good way to solve. I'm willing to take your advises. 
It's too late today, I'll continue this adventure tomorrow, :-)

**Comment by sokra** [comment]
Thanks

**Comment by zcfan** [comment]
@sokra Thank you for your patient :-)

**Pull request merged**